internal/volumes/mount.go

```diff
@@ -203,10 +207,39 @@ func (s *LinuxMountService) FormatDisk(disk string, fstype string) error {
 		_, _, err := command("mkfs.ext4", "-F", "-m0", disk)
 		return err
 	case "xfs":
-		_, _, err := command(
-			"mkfs.xfs",
-			"-i", "nrext64=0", // Compatibility with kernel that do not support nrext64
-			disk)
+		var err error
+		// By default node checks its own minimal supported kernel version
+		var xfsConfigPath = s.xfsConfigPath
+
+		// User configures cluster wide minimal supported kernel version
+		if xfsMinSupportedKernel != "" {
+			kv, err := ParseKernelVersion(xfsMinSupportedKernel)
+			if err == nil {
+				xfsConfigPath = GetXFSConfigPath(kv)
+			}
+		}
+
+		if xfsConfigPath != "" {
+			_, _, err = command(
+				"mkfs.xfs", "-f", "-c", fmt.Sprintf("options=%s", s.xfsConfigPath), disk)
+
+			if err == nil {
+				level.Info(s.logger).Log(
+					"msg", "formatted disk with xfs",
+					"xfs-config-path", xfsConfigPath,
+					"is-cluster-config", xfsMinSupportedKernel != "",
+				)
+			}
+		} else {
+			// Fallback
+			// Default flags extracted from /usr/share/xfsprogs/mkfs/lts_4.19.conf
```

Comment:
I think this should be documented: we only support Linux kernels that are not EOL, so >=4.19 as of today.

We should also mention that this kernel version policy is only related to XFS for now, and maybe that bumping the minimal Linux kernel version will not require a new major version?